src/main/java/org/candlepin/model/Consumer.java

```diff
@@ -1024,4 +988,32 @@ public Consumer setServiceType(String serviceType) {
         this.serviceType = serviceType;
         return this;
     }
+
+    public Map<String, String> getEnvironmentIds() {
+        return this.environmentIds;
```

Comment:
Let's use a MapView to wrap this map before returning it (see examples of how it's used in other entities like Product).
This will prevent people abusing this by doing `consumer.getEnvironmentIds().put("x", "y")`, while they should be using the `addEnvironment()` method instead

Reply:
Done.